src/cv/yolo_video_detect.py

```diff
@@ -0,0 +1,114 @@
+import cv2
+import imutils
+import numpy as np
+import os
+import time
+
+def get_configs(args):
+    """ Retrieves the object detector model resources """
+    labelsPath = os.environ.get('YOLO_NAMES_PATH')
+    labels = open(labelsPath).read().strip().split("\n")
+
+    np.random.seed(int(time.time()))
+    colors = np.random.randint(0, 255, size=(len(labels), 3), dtype="uint8")
+
+    weightsPath = os.environ.get('YOLO_WEIGHTS_PATH')
+    configPath = os.environ.get('YOLO_CONFIG_PATH')
+
+    return labels, colors, weightsPath, configPath
+
+def load_object_detector(configPath, weightsPath):
+    """ Loads the yolo model """
+    print("Loading Object Detector")
+    net = cv2.dnn.readNetFromDarknet(configPath, weightsPath)
+    ln = net.getLayerNames()
+    ln = [ln[i[0] - 1] for i in net.getUnconnectedOutLayers()]
+    return net, ln
+
+def init_video_stream(args):
+    """ Obtains the video stream """
+    vs = cv2.VideoCapture(args["input"])
+    try:
+        if imutils.is_cv2():
+            prop = cv2.cv.CV_CAP_PROP_FRAME_COUNT
+        else:
+            prop = cv2.CAP_PROP_FRAME_COUNT
+        total = int(vs.get(prop))
+        print("{} total frames in video".format(total))
+
+    except:
+        print("Could not determine # of frames in video")
+        total = -1
+
+    return vs, total
+
+def iterate_frames(args, vs, net, ln, colors, labels, total):
+    """ Iterate through each frame and pass it through the net """
+    writer = None
+    (W, H) = (None, None)
+    while True:
+        (grabbed, frame) = vs.read()
+        if writer is None:
+            fourcc = cv2.VideoWriter_fourcc(*"MJPG")
+            writer = cv2.VideoWriter(args["output"], fourcc, 30, (frame.shape[1], frame.shape[0]), True)
```

Comment:
There should be 50 to 80 characters per line

Reply:
modified